File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys
fi = sys.stdin.readline
                
def solution():
    t = int(fi().rstrip())
    n = list(map(int, fi().split()))
    x = int(fi().rstrip())
    
    n.sort()
    res = 0
    l = 0
    r = len(n) - 1
    
    while r - l > 0:
        if n[l] + n[r] == x:
            res += 1
            r = r - 1
        elif n[l] + n[r] < x:
            l = l + 1
        elif n[l] + n[r] > x:
            r = r - 1
    print(res)
        
    
if __name__ == '__main__':
    solution()

File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void solution() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        for(int i=0;i<n;i++) {
            int num = Integer.parseInt(st.nextToken());
            arr[i] = num;
        }
        Arrays.sort(arr);
        
        int x = Integer.parseInt(br.readLine());
        int l=0, r=arr.length-1, cnt=0;
        while (l<r) {
            if(arr[l]+arr[r]<x) l++;
            else if(arr[l]+arr[r]>x) r--;
            else if(arr[l]+arr[r]==x) {
                cnt++;
                l++;
                r--;
            }
        }
        sb.append(cnt);
        
        System.out.print(sb);
        br.close();
    }
    
    public static void main(String[] args) throws IOException {
        Main.solution();
    }
}